packages/eslint-plugin/src/rules/prefer-includes.ts

```diff
@@ -155,6 +155,21 @@ export default createRule({
           }
         }
 
+        // Check if `indexOf` is used as part of `includes` method declaration
+        let parent = node.parent;
```

Comment:
You should check if `MemberExpression#object` is `this` or not before this check. Because `this.array.indexOf()` should be reported.